main/src/ca/uwaterloo/flix/language/phase/unification/BoolAlg.scala

```diff
@@ -153,8 +153,17 @@ trait BoolAlg[F] {
     case Type.Apply(Type.Cst(TypeConstructor.Complement, _), tpe1, _) => mkNot(fromType(tpe1, env))
     case Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Union, _), tpe1, _), tpe2, _) => mkAnd(fromType(tpe1, env), fromType(tpe2, env))
     case Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Intersection, _), tpe1, _), tpe2, _) => mkOr(fromType(tpe1, env), fromType(tpe2, env))
-    case Type.Cst(TypeConstructor.Error(_, _), _) => mkTrue
+    case _ if containsError(t) => mkTrue
```

Comment:
@JonathanStarup Probably around here something else needs to happen?

(Not in this PR, but in your PR)